src/main/java/it/pn/frontend/e2e/pages/destinatario/personaGiuridica/UtentiPGPage.java

```diff
@@ -0,0 +1,349 @@
+        }
+    }
+
+    public void insertData(String codiceFiscale,String name, String surname, String email ) throws InterruptedException {
+
+        codiceFiscaleBox.sendKeys(codiceFiscale);
+        Thread.sleep(2000);
+        if (nameBox.getAttribute("value").equalsIgnoreCase(name) && surnameBox.getAttribute("value").equalsIgnoreCase(surname)){
+            logger.info("Il nome e il cognome è generato correttamente");
+        }else {
+            logger.error("Il nome e il cognome non è generato correttamente");
+            Assert.fail("Il nome e il cognome non è generato correttamente");
+        }
+
+        this.js().executeScript("arguments[0].setAttribute('autocomplete', 'off')" , emailBox);
+        emailBox.sendKeys(Keys.chord(Keys.CONTROL,"a"),Keys.DELETE);
+        emailBox.sendKeys(email);
+        confirmEmailBox.clear();
+        confirmEmailBox.sendKeys(email);
+    }
+
+    public void selectProduct(String product){
+        getWebDriverWait(30).withMessage("il combobox seleziona il prodotto non è cliccabile").until(ExpectedConditions.visibilityOf(selectProductDropdown));
+        actions.moveToElement(selectProductDropdown).click().perform();
+        WebElement productButton = driver.findElement(By.xpath("//li[contains(text(),'" + product + "')]"));
+        getWebDriverWait(30).withMessage("il prodotto" + product + "non è cliccabile").until(ExpectedConditions.elementToBeClickable(productButton));
+        productButton.click();
+        logger.info("Ruolo :" + product);
+        getWebDriverWait(30).withMessage("il radioBottone Amministratore della pagina aggiungi nuovo utente non è visibile").until(ExpectedConditions.visibilityOf(adminRadioButton));
+    }
+
+    public void selectRole(){
+        adminRadioButton.click();
+        logger.info("Si clicca sul bottone Continue");
+        actions.moveToElement(continueButton).click().perform();
+        getWebDriverWait(30).withMessage("il popup assegna ruolo non è visualizzata").until(ExpectedConditions.visibilityOf(confirmPopup));
+    }
+
+    public void clickRejectPopup(){
+        getWebDriverWait(30).withMessage("il bottone annula non è visibile").until(ExpectedConditions.visibilityOf(annulaButton));
+        logger.info("Si clicca sul bottone Annula");
+        annulaButton.click();
+    }
+
+    public void clickContinueAndAssign(){
+        continueButton.click();
+        getWebDriverWait(30).withMessage("il bottone assegna non è visibile").until(ExpectedConditions.visibilityOf(assegnaButton));
+        assegnaButton.click();
+    }
+
+    public void waitSuccessMessage(){
+        try {
+
+            if (successMessage.isDisplayed()) {
+                logger.info("Si visualizza correttamente messaggio di successo");
+            }
+        } catch (NoSuchElementException e ){
+            logger.error("Hai già aggiunto questo utente." + e);
+            Assert.fail("Hai già aggiunto questo utente." + e);
+        }
```

Comment:
You don't need to throw any exceptions; simply verify if the element exists and is displayed. Otherwise, it will fail as usual